core/storage/base_model/gae_models.py

```diff
@@ -530,6 +530,28 @@ def get_version(cls, entity_id, version_number):
         # pylint: enable=protected-access
 
     @classmethod
+    def get_multi_versions(cls, entity_id, version_numbers):
+        """Gets model instances for each version specified in version_numbers.
+
+        Args:
+            entity_id: str. ID of the entity.
+            version_numbers: list(int). List of version numbers.
+
+        Returns:
+            list(VersionedModel|None). Model instances representing the given
+                versions.
+        """
+        instances = []
+        # pylint: disable=protected-access
+        for version in version_numbers:
+            cls.get(entity_id)._require_not_marked_deleted()
```

Comment:
Why do this check here? The deleted status would be the same for all versions, right -- so we can do the check at the outset?

Reply:
Yep. Done. Checking at the outset now.